ui/app/components/keymgmt/provider-edit.js

```diff
@@ -95,7 +96,7 @@ export default class KeymgmtProviderEdit extends Component {
   async onDeleteKey(model) {
     try {
       await model.destroyRecord();
-      this.args.model.keys.removeObject(model);
+      this.args.model.keys = removeFromArray(this.args.model.keys, model);
```

Comment:
TODO: check if this is actually needed -- destroyRecord should automatically remove it from store/model

Reply:
Note: working on checking that this line isn't necessary, but we don't have a good mirage setup for keymgmt 😬 so it might take a while

Reply:
Update makes sense, in this case we're destroying a record which will automatically remove it from the store's list of records for that type. But generally yes, there are definitely reasons to reload the model after update particularly